### ISSUE TITLE
MoveTo doesn't work when a condition is "MoveTo is moving" evaluates as false

### ISSUE THREAD

**Issue by Scoremonger** [Scirra/Construct-3-bugs#6095]
## Problem description

I created an OR block using two conditions - `MoveTo is moving` and a comparison between two numbers. I added a `MoveTo: Move to position` action in the same block. I found that the `MoveTo: Move to position` never seemed to execute unless the "is moving" condition evaluated true. However, other actions in the same block **do** execute if at least one of the conditions in the OR block is true. I've added a text object to demonstrate this in the attached project.

## Attach a .c3p

[MoveToOrBlockBug.zip](https://github.com/Scirra/Construct-3-bugs/files/9541593/MoveToOrBlockBug.zip)

## Steps to reproduce

1. Create a sprite and add the MoveTo behavior to it.
2. Create a second sprite and place it some distance away from the first.
3. Add an event with the conditions `MoveTo is moving` and `System: Compare two values`, and leave the comparison at `0 = Equal to 0`, so it will evaluate true.
4. Make the conditions an OR block.
5. Add an action for the sprite with the MoveTo behavior: `MoveTo: Move to object` and set the object to the other sprite.
6. Run the project.

## Observed result

The sprite with the MoveTo behavior doesn't move.

## Expected result

Because one of the two OR conditions is true, the sprite should move.

## More details

**Affected browsers/platforms:** Chrome, Edge (Windows store version of Construct.)

**First affected release:** Tested as far back as r293.3 (probably happens before that), attached project is in r309 beta.

## System details

<details><summary>View details</summary>

Platform information
Product: Construct 3 r309 (beta)
Browser: Edge 105.0.1343.27
Browser engine: Chromium
Context: webapp
Operating system: Windows 11
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 20
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36 Edg/105.0.1343.27
Language setting: en-US

Local storage
Storage quota (approx): 1.1 tb
Storage usage (approx): 128 mb (0%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (NVIDIA)
Renderer: ANGLE (NVIDIA, NVIDIA GeForce RTX 3080 Direct3D11 vs_5_0 ps_5_0, D3D11)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_draw_buffers_indexed
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.265 (video/mp4; codecs=hev1.1.2.L93.B0)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
This is a regular misunderstanding of OR blocks.

The system condition 0 = 0 is always true, so the block always runs. However if the "MoveTo is moving" condition is false, then no instances meet the condition. The block will run with zero instances picked, and so its actions do nothing.

**Comment by WilsonPercival** [comment]
> This is a regular misunderstanding of OR blocks.
> 
> The system condition 0 = 0 is always true, so the block always runs. However if the "MoveTo is moving" condition is false, then no instances meet the condition. The block will run with zero instances picked, and so its actions do nothing.

@AshleyScirra But in classic it works!

![aaa](https://user-images.githubusercontent.com/91274932/189637918-3bf4f0f4-36c3-4674-8f1b-fcbf85926ba3.png)

**Comment by AshleyScirra** [comment]
Classic was retired a decade ago and should not be considered relevant to modern Construct.

**Comment by Scoremonger** [comment]
That is hideously unintuitive.

**Comment by Scoremonger** [comment]
@AshleyScirra But if I explicitly pick the sprite (with a For Each condition in a parent block, for example), it still doesn't work. Why would a condition effectively clear the instance that was picked? If I make an on tick event and use the same action with no condition, it works even though no object was explicitly picked. I would expect the same behavior in this case. I mean, what user could anticipate that an action will silently fail in an OR block after they have explicitly picked an object, but **will** work in a case where no object was explicitly picked? Surely this isn't the intended design?

**Comment by Scoremonger** [comment]
[ExplictlyPicked.zip](https://github.com/Scirra/Construct-3-bugs/files/9550821/ExplictlyPicked.zip)

Here's the same project with the instance explicitly picked in the parent block.

**Comment by AshleyScirra** [comment]
That doesn't change anything. The false 'Mover' condition means zero instances are picked for the actions. Normally the event would not run in that situation, but it's forced to run because it also says "OR 0 = 0", which is true, so the actions go ahead and run anyway. However as there are no 'Mover' instances that met the conditions then its actions will have no effect.

If the action did run then it would mean it ran an action on an instance that did *not* meet the conditions, which breaks the fundamental purpose of conditions.

**Comment by AshleyScirra** [comment]
I care enough to have tried to take the time to explain in detail what's going on. I understand this might not seem intuitive, but it is a consequence of the existing rules we've set up for other features. Changing the design of existing features is both outside the scope of bug reports (those tend to be either discussed on the forum or posted as feature requests), and changing something without thinking through the consequences in detail risks breaking thousands of existing projects, which is a far worse situation than we started with. So it's not necessarily easy to just do a quick fix in response to a user query.
